src/views/SignUp.vue

```diff
@@ -0,0 +1,166 @@
+            />
+            <v-text-field
+              id="confirm-password"
+              label="Confirm password"
+              name="confirmPassword"
+              prepend-icon="mdi-lock-check"
+              v-model="confirmPassword"
+              :append-icon="showConfirmPasswor ? 'mdi-eye' : 'mdi-eye-off'"
+              :type="showConfirmPasswor ? 'text' : 'password'"
+              :rules="[rules.required]"
+              :error="hasConfirmPwdError"
+              :error-messages="confirmPwdErrorMsgs"
+              @input="resetConfirmPassword()"
+              @click:append="showConfirmPasswor = !showConfirmPasswor"
+            />
+          </v-form>
+        </v-card-text>
+        <v-card-actions>
+          <v-btn text x-small link :to="{ name: 'login' }">
+            I am already registered back to login
+          </v-btn>
+          <v-spacer />
+          <v-btn
+            :light="$vuetify.theme.dark && valid"
+            :dark="!$vuetify.theme.dark && valid"
+            @click="handleSubmit"
+            :disabled="!valid"
+            >Sign Up</v-btn
+          >
+        </v-card-actions>
+      </v-card>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Ref } from "vue-property-decorator";
+import { namespace } from "vuex-class";
+
+import { CreateUser } from "../model/create-user";
+
+const auth = namespace("Auth");
+
+@Component
+export default class SignUp extends Vue {
+  private valid = false;
+  private alertMsg = "";
+  private user = "";
+  private preferredUsername = "";
+  private email = "";
+  private password = "";
+  private showPassword = false;
+  private confirmPassword = "";
+  private showConfirmPasswor = false;
+  private confirmPwdErrorMsgs: string[] = [];
+  private hasConfirmPwdError = false;
+
+  private rules = {
+    required: ($: string) => !!$ || "Required.",
+    preferredUsernameMin: ($: string) =>
+      $.length >= 3 ||
+      "This value is too short. It should have 3 characters or more.",
+    usernameMin: ($: string) =>
+      $.length >= 5 ||
+      "This value is too short. It should have 5 characters or more.",
+    min: ($: string) => $.length >= 8 || "Min 8 characters",
+    emailRules: ($: string) => /.+@.+\..+/.test($) || "E-mail must be valid",
+    hasLetterRules: ($: string) =>
+      /[A-Za-z]/.test($) || "Your password must contain at least one letter.",
+    hasNumberRules: ($: string) =>
+      /[0-9]/.test($) || "Your password must contain at least one digit.",
+  };
+
+  @Ref("signUpForm") readonly form!: HTMLFormElement;
+
+  @auth.Action
+  public createUser!: (user: CreateUser) => Promise<void>;
+
+  public handleSubmit() {
+    if (this.password !== this.confirmPassword) {
+      this.confirmPwdErrorMsgs.push("Doesn't match Password");
+      this.hasConfirmPwdError = true;
+      return;
+    }
+    if (this.form.validate()) {
+      this.createUser({
+        username: this.user,
+        displayName: "asfsa",
```

Comment:
what's this? 😅 I wouldn't give the use an option to define their displayName upon sign up. A user can change it later by editing their profile. Or is it required for signup?

Reply:
:see_no_evil: